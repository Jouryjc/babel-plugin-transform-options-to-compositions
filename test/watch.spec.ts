import plugin from '../lib/index'
import { transformSync } from '@babel/core'

describe('watch -> watch', () => {
  test('watch', () => {
    const { code } = transformSync(
      `
      export default {
        watch: {
          name: {
            deep: true,
            handler (cur, prev) {
              console.log(cur, prev)
            }
          }
        },
      }
    `,
      {
        presets: [
          [
            '@babel/preset-env',
            {
              modules: false,
            },
          ],
        ],
        plugins: [plugin],
      }
    )

    expect(code).toMatchSnapshot('watch')
  })
})
