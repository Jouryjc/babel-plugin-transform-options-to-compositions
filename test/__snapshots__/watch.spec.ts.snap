// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`watch -> watch watch: watch 1`] = `
"import { watch } from 'vue';
watch('name', (cur, prev) => {
  console.log(cur, prev);
}, {
  deep: true
});"
`;
